plugins/woocommerce/tests/e2e-pw/tests/merchant/create-simple-product.spec.js

```diff
@@ -155,6 +148,14 @@ test.describe( 'Add New Simple Product Page', () => {
 		await expect(
 			page.locator( 'a.shipping-calculator-button' )
 		).toBeVisible();
-		await page.click( 'a.remove' );
+		await page.click(
+			`a.remove[data-product_id='${ nonVirtualProductId }']`
+		);
+		await page.waitForLoadState( 'networkidle' );
+		await expect(
+			page.locator(
+				`a.remove[data-product_id='${ nonVirtualProductId }']`
+			)
+		).not.toBeVisible();
```

Comment:
Adding a wait and assertion, same reasoning as mentioned [here](https://github.com/woocommerce/woocommerce/pull/35760/files#r1042011250).